delfin/drivers/hpe/hpe_msa/ssh_handler.py

```diff
@@ -479,3 +478,319 @@ def handle_xml_to_dict(xml_info, element):
+            }
+            if host_groups_json:
+                for initiator in host_groups_json:
+                    description = type_switch.get(
+                        initiator.get('host-bus-type-numeric'),
+                        consts.InitiatorType.UNKNOWN_INITIATOR_DESCRIPTION)
+                    initiator_item = {
+                        "name": initiator.get('nickname'),
+                        "type": description,
+                        "alias": initiator.get('durable-id'),
+                        "storage_id": storage_id,
+                        "native_storage_host_initiator_id":
+                            initiator.get('durable-id'),
+                        "wwn": initiator.get('id'),
+                        "status": constants.InitiatorStatus.ONLINE,
+                        "native_storage_host_id": initiator.get('host-id')
+                    }
+                    initiator_list.append(initiator_item)
+            return initiator_list
+        except Exception as e:
+            LOG.error("Failed to get initiators from msa")
+            raise e
+
+    def list_storage_hosts(self, storage_id):
+        try:
+            hosts_info = self.ssh_pool.do_exec('show host-groups')
+            host_list = []
+            hosts = self.handle_xml_to_json(hosts_info, 'host')
+            host_map = {}
+            for host in hosts:
+                status = constants.HostStatus.NORMAL
+                os_type = constants.HostOSTypes.HP_UX
+                host_member_count = int(host.get('member-count'))
+                if host_member_count > 0:
+                    serial_number = host_map.get(host.get('serial-number'))
+                    if not serial_number:
+                        host_map[host.get('serial-number')] = \
+                            host.get('serial-number')
+                        host_dict = {
+                            "name": host.get('name'),
+                            "description": host.get('durable-id'),
+                            "storage_id": storage_id,
+                            "native_storage_host_id":
+                                host.get('serial-number'),
+                            "os_type": os_type,
+                            "status": status
+                        }
+                        host_list.append(host_dict)
+            return host_list
+        except Exception as e:
+            LOG.error("Failed to get host from msa")
+            raise e
+
+    def list_storage_host_groups(self, storage_id):
+        try:
+            host_groups_info = self.ssh_pool.do_exec('show host-groups')
+            storage_host_groups_result = {}
+            host_group_list = []
+            storage_host_grp_relation_list = []
+            host_groups = self.handle_xml_to_json(
+                host_groups_info, 'host-group')
+            host_info_list = self.handle_xml_to_json(host_groups_info, 'host')
+            if host_groups:
```

Comment:
I think `if host_groups:` is unnecessary, because `handle_xml_to_json` will return `[]` if there is no data in `host_groups_info` .

Reply:
Deleted this judgment
![image](https://user-images.githubusercontent.com/88076879/155675636-04d4c7fc-af29-4870-8d8e-3de4e64736da.png)